### PR TITLE
Use antsibull 0.42.0b1 for docs build sanity test and docs build

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
So that @bcoca's https://github.com/ansible/ansible/pull/74963 can pass tests and get merged.

@samccann needs to build the test docsite with this first so we can check whether there is no unintentional breakage.

##### ISSUE TYPE
- Docs Pull Request
- Test Pull Request

##### COMPONENT NAME
CI
docsite

### CODE REVIEW

File: docs/docsite/known_good_reqs.txt
Comment (mattclay) [inline docs/docsite/known_good_reqs.txt:7]: Why a beta release? Is this just for testing in this PR before merging with a stable version?

Comment (felixfontein) [reply docs/docsite/known_good_reqs.txt:7]: Because it contains a minimal fix needed to unblock #74963. A better fix (resp. a proper implementation for ansible-core 2.13+ which stops using internal APIs) requires #77035 to be merged, but I was told this will not happen before #74963 is merged.

**Review by samccann — APPROVED** [review]
Verified the output at http://docs.testing.ansible.com/ansible/devel/index.html

Leaving to someone else to address the beta comment before merging

**Comment by felixfontein** [comment]
Closing since this is now included in #74963. @samccann @mattclay thanks for your reviews!

**Pull request closed without merging**
